server/src/main/java/com/octopus/teamcity/opentelemetry/server/helpers/OTELHelperImpl.java

```diff
@@ -77,4 +76,10 @@ public void addAttributeToSpan(Span span, String attributeName, Object attribute
         LOG.debug("Adding attribute to span " + attributeName + "=" + attributeValue);
         span.setAttribute(attributeName, attributeValue.toString());
     }
+
+    @Override
+    public void release() {
+        this.sdkTracerProvider.forceFlush();
```

Comment:
Can we check here to close the `openTelemetry` `tracer` and nullify the `spanMap` here?

Reply:
The `tracer` has no close (or related options). I considered setting it to `null`, but the field is `final`, an this whole class should be garbage collected, so it shouldn't be necessary.
The `spanMap`, I've added a `.clear()`, which probably isn't strictly needed, as it should GC itself, but hey, why not.